interfaces/builtin/hidraw.go

```diff
@@ -172,14 +172,11 @@ func (iface *hidrawInterface) UDevConnectedPlug(spec *udev.Specification, plug *
 		return nil
 	}
 
-	for appName := range plug.Apps {
-		tag := udevSnapSecurityName(plug.Snap.Name(), appName)
-		if hasOnlyPath {
-			spec.AddSnippet(fmt.Sprintf("SUBSYSTEM==\"hidraw\", KERNEL==\"%s\", TAG+=\"%s\"", strings.TrimPrefix(path, "/dev/"), tag))
-
-		} else {
-			spec.AddSnippet(udevUsbDeviceSnippet("hidraw", usbVendor, usbProduct, "TAG", tag))
-		}
+	if hasOnlyPath {
+		spec.TagDevice(fmt.Sprintf(`SUBSYSTEM=="hidraw", KERNEL=="%s"`, strings.TrimPrefix(path, "/dev/")))
+	} else {
+		spec.TagDevice(fmt.Sprintf(`IMPORT{builtin}="usb_id"
+SUBSYSTEM=="hidraw", SUBSYSTEMS=="usb", ATTRS{idVendor}=="%04x", ATTRS{idProduct}=="%04x"`, usbVendor, usbProduct))
```

Comment:
I'm not totally sure it's more readable than calling udevUsbDeviceSnippet helper here to be honest. Anyway, not biggie but if you decide not to use udevUsbDeviceSnippet, then perhaps udevUsbDeviceSnippet can be removed from utils?

Reply:
I agree, either use udevUsbDeviceSnippet where you can or remove it from utils.go.

Reply:
I dropped it entirely now. I think the result is much more readable.